pkg/planner/core/expression_rewriter.go

```diff
@@ -2676,3 +2690,21 @@ func hasCurrentDatetimeDefault(col *model.ColumnInfo) bool {
 	}
 	return strings.ToLower(x) == ast.CurrentTimestamp
 }
+
+// hasLimit checks if the plan already contains a LIMIT operator
+func (er *expressionRewriter) hasLimit(plan base.LogicalPlan) bool {
```

Comment:
this func is not er related, can we just impl it as a simple func without receiver